### DISCUSSION TITLE
How do you configure the data/time formats in the chat windows?

### DISCUSSION THREAD

**Discussion by pdwalker** [d99kris/nchat#162] [Q&A]
In your sample screenshots, you have different date formats presented
https://github.com/d99kris/nchat/blob/master/doc/screenshot-nchat.png shows "YYYY-MM-DD HH:MM" while https://github.com/d99kris/nchat/blob/master/doc/screenshot-nchat-dracula.png shows it as "DD MMM YYYY HH:MM"

When I run the program, I get "HH:MM"

I was not able to find anything in the documentation on how the date format is determined.  Could you tell me how I can change the date/time presentation in the ui?

Much appreciated.

**Comment by d99kris** [comment]
Hi @pdwalker - the date format is currently not configurable. The format is dynamic with the intention to provide shortest unambiguous timestamp (similar to how one communicates time in spoken language). More specifically:
`HH:MM` for timestamps on same date as today, e.g. `19:00`
`DAY HH:MM` for timestamps in the last week, e.g. `Mon 19:00`
`DD MMM HH:MM` for timestamps in the current year, e.g. `14 Nov 19:00`
`DD MMM YYYY HH:MM` for timestamps in non-current year, e.g. `14 Nov 2022 19:00`

The screenshots are possibly a bit outdated, I'll see if I can update them some day.

Are you looking for any particular type of timestamp? If you for example prefer to always get full timestamp `DD MMM YYYY HH:MM` we could probably add an option for that, but I hope not to need too fancy configurability of the timestamps. 🙂

**Comment by pdwalker** [comment]
Understood.

Being able to specify the timestamp format would be a "nice to have" feature, but don't bust a gut over it.  Now that I know how you dynamically change the date format, I can work with it.  

Thanks for answering.

**Reply by d99kris** [reply]
Cool, btw if you had a specific time stamp format in mind, feel free to share it, it could be it's a simple thing to enable (like `YYYY-MM-DD HH:MM` exists in the code and would just require me to add a flag setting to enable). In any case it can be a good datapoint. Chances are there are more users with similar needs.

**Reply by pdwalker** [reply]
My preference is ```YYYY-MM-DD HH:MM``` (leading 0s, 24H clock) as it sorts properly as text and is unambiguous.

**Reply by d99kris** [reply]
> My preference is `YYYY-MM-DD HH:MM` (leading 0s, 24H clock) as it sorts properly as text and is unambiguous.

I think that's reasonable to support, I've implemented it now in 420dd5c - you'll need to edit `~/.nchat/app.conf` and set
```
timestamp_iso=1
```
in order to enable it.
